clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -633,9 +652,10 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
      * @param response The produce response
      * @param correlationId The correlation id for the request
      * @param now The current POSIX timestamp in milliseconds
+     * @param partitionsWithUpdatedLeaderInfo This will be populated with partitions that have updated leader info.
      */
     private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionResponse response, long correlationId,
-                               long now) {
+                               long now, Map<TopicPartition, Metadata.LeaderIdAndEpoch> partitionsWithUpdatedLeaderInfo) {
```

Comment:
Can we wrap this `Map` in an `Optional` or explicitly allow for `null` since four out of the five callers aren't interested in the updates to the map?